DEVELOPERS.md

```diff
@@ -28,138 +30,75 @@ jQuery for interactivity.
+compiler, a Rust compiler, and various libraries.  [There is
+documentation listing the commands you can use for each
 platform](https://github.com/rbenv/ruby-build/wiki#suggested-build-environment)
 (e.g. Macintosh Homebrew, or Ubuntu).
 
-After that, install Ruby.  `asdf` will consult the `.tool-versions`
-file to select the version.
-
-```sh
-$ asdf install ruby
-```
-
-Having done this, you can see if your `$PATH` finds the `ruby` "shim"
-generated by `asdf` and consult the version:
-
-```sh
-$ which ruby
-/home/ubicloud/.asdf/shims/ruby
-$ ruby --version
-ruby 3.2.5 (2024-07-26 revision 31d0f1a2e7) [x86_64-linux]
-```
-
-### Installing asdf-direnv
-
-We find use of `asdf` with
-[asdf-direnv](https://github.com/asdf-community/asdf-direnv) almost
-obligatory, for the reasons discussed in its README.  Let's set it up
-as a user-global tool, and not a project-local one:
-
-```sh
-$ asdf direnv setup --version latest
-$ echo "direnv $(direnv --version)" >> ~/.tool-versions
-```
-
-After `direnv setup` you need to source your shell's startup files or
-start a new shell.
-
-Now, upon `cd`-ing into the clover directory -- which already has a
-[.envrc](.envrc) committed -- you should see something like:
+You will need
+[dependencies](https://github.com/mise-plugins/mise-postgres#dependencies)
+to compile Postgres installed on your system.
 
-```sh
-direnv: error /home/ubicloud/code/clover/.envrc is blocked. Run `direnv allow` to approve its content
-```
 
-Okay, let's allow direnv in this directory:
+After that, run `mise install`. `mise` will consult the
+`.tool-versions` file to select the version:
 
 ```sh
-$ asdf exec direnv allow .
-direnv: loading ~/code/clover/.envrc
-direnv: using asdf
-direnv: Creating env file /home/ubicloud/.cache/asdf-direnv/env/2363097900-478416608-3909218245-3753665172
-[...]
+$ mise install
 ```
 
-You should still be able to get the same Ruby version:
-
-```sh
-$ ruby --version
-ruby 3.2.5 (2024-07-26 revision 31d0f1a2e7) [x86_64-linux]
-```
-
-But, instead of being resolved through a "shim" program, the binary is
-referenced directly:
+Having done this, you can see if your `$PATH` finds the mise-installed
+`ruby`:
 
 ```sh
 $ which ruby
-/home/ubicloud/.asdf/installs/ruby/3.2.5/bin/ruby
+/home/youruser/.local/share/mise/installs/ruby/3.2.7/bin/ruby
+$ ruby --version
+ruby 3.2.7 (2025-02-04 revision 02ec315244) [x86_64-linux]
 ```
 
 ### Installing Postgres
 
 You will need
-[dependencies](https://github.com/smashedtoatoms/asdf-postgres#dependencies)
+[dependencies](https://github.com/mise-plugins/mise-postgres#dependencies)
 to compile Postgres installed on your system.
 
-First, set some autoconf `./configure` options to be passed to
-Postgres:
-
-```sh
-$ echo "POSTGRES_EXTRA_CONFIGURE_OPTIONS='--with-uuid=e2fs --with-openssl'" > ~/.asdf-postgres-configure-options
-```
-
 Then run:
 
 ```sh
-$ asdf install postgres
+$ mise install postgres
 ```
 
 There are many alternative ways to get Postgres, e.g. via system
```

Comment:
> There are many alternative ways to get Postgres, e.g. via system
package manager, Homebrew, Postgres.app, etc.  They are all acceptable, our version requirements for Postgres are more relaxed than with Ruby.

Since we have `postgres 15.8` in our `.tool-versions` file, running `asdf/mise install` will also install PostgreSQL. If we don't install it, a missing package warning will appear every time you enter the directory. If we really want to allow any PostgreSQL installation, we should remove it from the project's `.tool-versions` file and recommend adding it to the global `.tool-versions` file. Another option is to require the use of asdf/mise PG :D

Reply:
what do you think? On the one hand, it's semi-convenient that it just takes care of itself...kind of. If I remember right, though, some issues how the asdf plugin managed the data directory were tedious.

On the downside, it may be rather weird to quite a few people in a way that using `mise` or a similar program for handling ruby, python, go, etc is not.

maybe I will use the `mise.local.toml` feature to do this, and have a version controlled template file. I will probably symlink something like `mise.local.toml.template` to `mise.local.toml` and leave it. This is not quite as nice for people that want to use the mise.local.toml feature with more tools, but I think we can deal with that then.

Reply:
@enescakir done that way for now at least.